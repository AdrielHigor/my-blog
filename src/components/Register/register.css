form#Register {
  width: 100%;
  max-width: 450px;
  margin: 30px auto 0;
  padding: 30px;
  background: #ffffff;
  border: 1px solid #dddddd;
  display: flex;
  flex-direction: column;
}

form#Register input{
    margin-bottom: 15px;
    height: 40px;
    border-radius: 4px;
    border: 1px solid #ddd;
    padding: 0 20px;
    font-size: 14px;
}

form#Register button{
    padding:10px 20px;
    border-radius: 4px;
    background: #25282a;
    color: #ffffff;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
}

form#Register a, p{
    text-align: center;
    text-decoration: none;
    font-size: 16px;
    padding-top: 20px;
}

form#Register a{
    font-weight: bold;
}