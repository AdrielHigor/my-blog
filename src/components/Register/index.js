import "./register.css";
import React, { useEffect } from "react";
import { withRouter } from "react-router-dom";
import firebase from "../../firebase";


export default withRouter(function Register(props) {
    const [name, setName] = React.useState();
    const [email, setEmail] = React.useState();
    const [password, setPassword] = React.useState();

    useEffect(() => {
        if (firebase.getCurrent()) {
            props.history.replace('/dashboard')
        }
    })

    const register = React.useCallback(
        async e => {
            e.preventDefault();

            try {
                await firebase.register(name, email, password);
                props.history.replace("/dashboard");
            } catch (error) {
                alert(error.message);
            }
        },
        [name, email, password]
    );

    return (
        <div>
            <form onSubmit={register} id="Register">
                <label>Nome:</label>
                <input
                    type="text"
                    value={name}
                    onChange={e => setName(e.target.value)}
                    autoFocus
                    placeholder="Nome de usuÃ¡rio"
                />
                <label>Email:</label>
                <input
                    type="email"
                    value={email}
                    onChange={e => setEmail(e.target.value)}
                    placeholder="email@email.com"
                />
                <label>Senha:</label>
                <input
                    type="password"
                    value={password}
                    onChange={e => setPassword(e.target.value)}
                    placeholder="*******"
                />
                <button type="submit">Cadastrar</button>
            </form>
        </div>
    );
});
