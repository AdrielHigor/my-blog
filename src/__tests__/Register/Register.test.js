import React from "react";
import Register from "../../components/Register";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from "enzyme";

configure({ adapter: new Adapter() });

describe("Register Test Suite", () => {
    test("Register rendering correctly", () => {
        const component = shallow(<Register.WrappedComponent />)
        expect(component.html()).toMatchSnapshot();
    })

    test("Register submit form correctly", () => {
        let component

        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["Test User", jest.fn()])
            .mockImplementationOnce(() => ["testemail@teste.com", jest.fn()])
            .mockImplementationOnce(() => ["passw0rd", jest.fn()])

        const mockRegister = jest
            .spyOn(React, 'useCallback')
            .mockImplementation(() => { })

        component = shallow(<Register.WrappedComponent />);

        component.find("button").simulate("click");

        expect(mockRegister).toHaveBeenCalled();
    })
})