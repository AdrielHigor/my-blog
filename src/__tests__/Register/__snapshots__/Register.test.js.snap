// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Test Suite Register rendering correctly 1`] = `"<div><form id=\\"Register\\"><label>Nome:</label><input type=\\"text\\" autofocus=\\"\\" placeholder=\\"Nome de usuÃ¡rio\\"/><label>Email:</label><input type=\\"email\\" placeholder=\\"email@email.com\\"/><label>Senha:</label><input type=\\"password\\" placeholder=\\"*******\\"/><button type=\\"submit\\">Cadastrar</button></form></div>"`;
